crates/rust-analyzer/src/global_state.rs

```diff
@@ -63,7 +63,7 @@ pub(crate) struct GlobalState {
     req_queue: ReqQueue,
     pub(crate) task_pool: Handle<TaskPool<Task>, Receiver<Task>>,
     pub(crate) loader: Handle<Box<dyn vfs::loader::Handle>, Receiver<vfs::loader::Message>>,
-    pub(crate) flycheck: Option<FlycheckHandle>,
+    pub(crate) flycheck: Vec<FlycheckHandle>,
```

Comment:
I wonder at what level of abstraction this `Vec` belongs. I *think* long term it's better to move these process-herding into `FlycheckHandle`, so that the logic for "which flycheck to restrat" could be centralized there, but the current soln is fine. 

An interesting theoretical argument for pushing the Vec down:

With the current setup, we have *sequential* shutdown behavior.  If we have `n` flychecks, we kill & wait for them sequentially. The ideal behavior would be to first kill all of the processes, and then wait for them in parallel.